source/workers/LocalWorker.cpp

```diff
@@ -3329,7 +3333,66 @@ void LocalWorker::s3ModeIterateBuckets()
 			}
 		}
 
-		// delete buckets
+//        TODO:
+//         1. add pseudo random tag names:  seed(n):  k"short_tag": "min chars", "max
+//              1. min length key and value
+//              2. max length key and value  (current max length of key: 128, value: 256)
+//              3. medium length key and value
+//         2. test with vast
+//         3. remove debug logs
+//         4. review with Sven
+        if(progArgs->getUseS3Metadata())
+        {
+            // PUT bucket tagging
+            const auto &bucket = bucketVec[bucketIndex];
+            S3::PutBucketTaggingRequest request;
```

Comment:
would be good to move the s3 request/response into a separate function. create/delete of buckets are admittedly bad examples for this that i should also move into a separate funtion. LocalWorker::s3ModeStatObject() and hopefully most other s3 requests are good examples for having the s3 request/response in a separate funtion :-)